### PR TITLE
fix: ContextualMenu - deprecate componentRef

### LINKED ISSUE #24716: [Bug]: ContextualMenu's componentRef is not initialised anymore after v8.36.3
### Library

React / v8 (@fluentui/react)

### System Info

```shell
System:
  OS: Windows 10 10.0.19044
  CPU: (20) x64 12th Gen Intel(R) Core(TM) i7-12800H
  Memory: 10.20 GB / 31.68 GB
Browsers:
  Edge: Spartan (44.19041.1266.0), Chromium (105.0.1343.27)
  Internet Explorer: 11.0.19041.1566
```

### Are you reporting Accessibility issue?

no

### Reproduction

https://codesandbox.io/s/sleepy-hugle-29121h

## Actual Behavior
When using the `componentRef` prop it never gets initialised (always null).

## Expected Behavior
The `componentRef` prop should be initialised with the component instance (otherwise the prop should be removed)

## Additional info
We use the `DefaultButton` component and set its `menuProps` object to render a `ContextualMenu` component (like you similarly do with your examples here https://developer.microsoft.com/en-us/fluentui#/controls/web/contextualmenu). 
We were using v. **8.19.1** and recently updated it the its latest version and after that the `componentRef` to the `ContextualMenu` instance never held a value anymore (we used it to dismiss the menu clicking on a custom element). From what I could see the first version with the issue is v8.36.4 while v.8.36.3 still held the ref correctly.

You can have a look at both samples here:
**v8.36.4** sandbox --> https://codesandbox.io/s/sleepy-hugle-29121h
**v.8.36.3** sandbox --> https://codesandbox.io/s/admiring-panna-nmdj2l

The samples are the exact same of yours (Basic and Default ContextualMenu samples): you click on the button, the menu opens and then click on the "New" item. You can see in the console the ref being logged correctly for v8.36.3 (for both samples) while nothing is logged with v8.36.4 (ref is null for both).

I had a look at the code and the `ContextualMenu` component was refactored in version 8.36.4 when the `ContextualMenuInternal` class was removed which was where the `componentRef` was actually initialiased. However, that initialisation seems to happen nowhere in code now (and if that's the case maybe a solution like the `useComponentRef` of the `Checkbox.Base` component might work? otherwise it looks like a breaking change to me).

### Logs

_No response_

### Requested priority

High

### Products/sites affected

_No response_

### Are you willing to submit a PR to fix?

no

### Validations

- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.
- [X] The provided reproduction is a minimal reproducible example of the bug.

### PR DESCRIPTION
**Pull request by micahgodbolt**
fixes #24716

ContextualMenu has no public methods, so the `componentRef` prop should have been deprecated when switching over to functional component. 

Alternatively, we could export a ref with nothing on it, but this better communicates how unnecessary the prop is.

### CODE REVIEW

**Review by spmonahan — APPROVED** [review]

**Review by sopranopillow — APPROVED** [review]

File: packages/react/src/components/ContextualMenu/ContextualMenu.base.tsx [hunk 1]
```diff
@@ -708,7 +709,9 @@ function useMouseHandlers(
 export const ContextualMenuBase: React.FunctionComponent<IContextualMenuProps> = React.memo(
   React.forwardRef<HTMLDivElement, IContextualMenuProps>((propsWithoutDefaults, forwardedRef) => {
     const { ref, ...props } = getPropsWithDefaults(DEFAULT_PROPS, propsWithoutDefaults);
-    const hostElement = React.useRef<HTMLDivElement>(null);
+    const rootRef = React.useRef<HTMLDivElement>(null);
+    const hostElement = useMergedRefs(forwardedRef, rootRef);
```
Comment (smhigley) [inline packages/react/src/components/ContextualMenu/ContextualMenu.base.tsx:713]: `forwardedRef` is used as the ref for the `<Callout>`, while `hostElement` is used as the ref for the child menu of the Callout. I'm not sure why those are the way they are, but I don't think they should be merged 😅

**Review by smhigley — CHANGES_REQUESTED** [review]

**Comment by fabricteam** [comment]
## 📊 Bundle size report

<sub>🤖 This report was generated against <a href='https://github.com/microsoft/fluentui/commit/3d828549c9e322ab46fdd80e30a29b037cfddb19'>3d828549c9e322ab46fdd80e30a29b037cfddb19</a></sub>
<div id="bundleSizeComment9423"/>

**Comment by fabricteam** [comment]
## [Perf Analysis (`@fluentui/react`)](https://github.com/microsoft/fluentui/wiki/Perf-Testing)

  <table>
  <tr>
    <th>Scenario</th>
    <th>Render type</th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">Master Ticks</a>
    </th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">PR Ticks</a>
    </th>
    <th>Iterations</th>
    <th>Status</th>
  </tr><tr>
            <td>ThemeProvider</td>
            <td>virtual-rerender-with-unmount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25499/perf-test/ThemeProvider-virtual-rerender-with-unmount_base.html">2138</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25499/perf-test/ThemeProvider-virtual-rerender-with-unmount.html">2129</a></td>
            <td>5000</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25499/perf-test/ThemeProvider-virtual-rerender-with-unmount.regression.txt">Possible regression</a></td>
           </tr></table><details><summary>All results</summary><p>
  <table>
  <tr>
    <th>Scenario</th>
    <th>Render type</th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">Master Ticks</a>
    </th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">PR Ticks</a>
    </th>
    <th>Iterations</th>
    <th>Status</th>
  </tr><tr>
            <td>BaseButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25499/perf-test/BaseButton-mount_base.html">1188</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25499/perf-test/BaseButton-mount.html">1175</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Breadcrumb</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25499/perf-test/Breadcrumb-mount_base.html">3045</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25499/perf-test/Breadcrumb-mount.html">3103</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>Checkbox</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25499/perf-test/Checkbox-mount_base.html">2615</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25499/perf-test/Checkbox-mount.html">2649</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>CheckboxBase</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25499/perf-test/CheckboxBase-mount_base.html">2400</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25499/perf-test/CheckboxBase-mount.html">2387</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>ChoiceGroup</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25499/perf-test/ChoiceGroup-mount_base.html">4435</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25499/perf-test/ChoiceGroup-mount.html">4161</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>ComboBox</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25499/perf-test/ComboBox-mount_base.html">1531</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25499/perf-test/ComboBox-mount.html">1223</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>CommandBar</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25499/perf-test/CommandBar-mount_base.html">10400</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25499/perf-test/CommandBar-mount.html">9860</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>ContextualMenu</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25499/perf-test/ContextualMenu-mount_base.html">11584</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25499/perf-test/ContextualMenu-mount.html">12527</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>DefaultButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25499/perf-test/DefaultButton-mount_base.html">1395</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25499/perf-test/DefaultButton-mount.html">1401</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>DetailsRow</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25499/perf-test/DetailsRow-mount_base.html">3552</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25499/perf-test/DetailsRow-mount.html">3433</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>DetailsRowFast</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25499/perf-test/DetailsRowFast-mount_base.html">3505</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25499/perf-test/DetailsRowFast-mount.html">3412</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>DetailsRowNoStyles</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25499/perf-test/DetailsRowNoStyles-mount_base.html">3428</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25499/perf-test/DetailsRowNoStyles-mount.html">7151</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Dialog</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25499/perf-test/Dialog-mount_base.html">2840</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25499/perf-test/Dialog-mount.html">3049</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>DocumentCardTitle</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25499/perf-test/DocumentCardTitle-mount_base.html">545</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25499/perf-test/DocumentCardTitle-mount.html">582</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>Dropdown</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25499/perf-test/Dropdown-mount_base.html">3201</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25499/perf-test/Dropdown-mount.html">3092</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FocusTrapZone</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25499/perf-test/FocusTrapZone-mount_base.html">2014</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25499/perf-test/FocusTrapZone-mount.html">1974</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FocusZone</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25499/perf-test/FocusZone-mount_base.html">1825</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25499/perf-test/FocusZone-mount.html">1942</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>GroupedList</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25499/perf-test/GroupedList-mount_base.html">1912</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25499/perf-test/GroupedList-mount.html">2094</a></td>
            <td>2</td>
            <td></td>
           </tr>
<tr>
            <td>GroupedList</td>
            <td>virtual-rerender</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25499/perf-test/GroupedList-virtual-rerender_base.html">1138</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25499/perf-test/GroupedList-virtual-rerender.html">1132</a></td>
            <td>2</td>
            <td></td>
           </tr>
<tr>
            <td>GroupedList</td>
            <td>virtual-rerender-with-unmount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25499/perf-test/GroupedList-virtual-rerender-with-unmount_base.html">1621</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25499/perf-test/GroupedList-virtual-rerender-with-unmount.html">1610</a></td>
            <td>2</td>
            <td></td>
           </tr>
<tr>
            <td>GroupedListV2</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25499/perf-test/GroupedListV2-mount_base.html">547</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25499/perf-test/GroupedListV2-mount.html">560</a></td>
            <td>2</td>
            <td></td>
           </tr>
<tr>
            <td>GroupedListV2</td>
            <td>virtual-rerender</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25499/perf-test/GroupedListV2-virtual-rerender_base.html">552</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25499/perf-test/GroupedListV2-virtual-rerender.html">537</a></td>
            <td>2</td>
            <td></td>
           </tr>
<tr>
            <td>GroupedListV2</td>
            <td>virtual-rerender-with-unmount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25499/perf-test/GroupedListV2-virtual-rerender-with-unmount_base.html">563</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25499/perf-test/GroupedListV2-virtual-rerender-with-unmount.html">536</a></td>
            <td>2</td>
            <td></td>
           </tr>
<tr>
            <td>IconButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25499/perf-test/IconButton-mount_base.html">1854</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25499/perf-test/IconButton-mount.html">1791</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Label</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25499/perf-test/Label-mount_base.html">690</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25499/perf-test/Label-mount.html">732</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Layer</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25499/perf-test/Layer-mount_base.html">4215</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25499/perf-test/Layer-mount.html">4095</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Link</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25499/perf-test/Link-mount_base.html">796</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25499/perf-test/Link-mount.html">756</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>MenuButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25499/perf-test/MenuButton-mount_base.html">1654</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25499/perf-test/MenuButton-mount.html">1566</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>MessageBar</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25499/perf-test/MessageBar-mount_base.html">2207</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25499/perf-test/MessageBar-mount.html">2309</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Nav</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25499/perf-test/Nav-mount_base.html">3222</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25499/perf-test/Nav-mount.html">3143</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>OverflowSet</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25499/perf-test/OverflowSet-mount_base.html">1368</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25499/perf-test/OverflowSet-mount.html">1363</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Panel</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25499/perf-test/Panel-mount_base.html">2328</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25499/perf-test/Panel-mount.html">2386</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>Persona</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25499/perf-test/Persona-mount_base.html">1260</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25499/perf-test/Persona-mount.html">1211</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>Pivot</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25499/perf-test/Pivot-mount_base.html">1640</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25499/perf-test/Pivot-mount.html">1560</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>PrimaryButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25499/perf-test/PrimaryButton-mount_base.html">1494</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25499/perf-test/PrimaryButton-mount.html">1434</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Rating</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25499/perf-test/Rating-mount_base.html">7096</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25499/perf-test/Rating-mount.html">6827</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>SearchBox</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25499/perf-test/SearchBox-mount_base.html">1519</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25499/perf-test/SearchBox-mount.html">1485</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Shimmer</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25499/perf-test/Shimmer-mount_base.html">2848</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25499/perf-test/Shimmer-mount.html">2837</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Slider</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25499/perf-test/Slider-mount_base.html">2159</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25499/perf-test/Slider-mount.html">2118</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>SpinButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25499/perf-test/SpinButton-mount_base.html">4668</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25499/perf-test/SpinButton-mount.html">4639</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Spinner</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25499/perf-test/Spinner-mount_base.html">807</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25499/perf-test/Spinner-mount.html">801</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>SplitButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25499/perf-test/SplitButton-mount_base.html">3050</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25499/perf-test/SplitButton-mount.html">3022</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Stack</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25499/perf-test/Stack-mount_base.html">823</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25499/perf-test/Stack-mount.html">852</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>StackWithIntrinsicChildren</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25499/perf-test/StackWithIntrinsicChildren-mount_base.html">2261</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25499/perf-test/StackWithIntrinsicChildren-mount.html">2471</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>StackWithTextChildren</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25499/perf-test/StackWithTextChildren-mount_base.html">4645</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25499/perf-test/StackWithTextChildren-mount.html">4669</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>SwatchColorPicker</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25499/perf-test/SwatchColorPicker-mount_base.html">9974</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25499/perf-test/SwatchColorPicker-mount.html">10603</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>TagPicker</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25499/perf-test/TagPicker-mount_base.html">2678</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25499/perf-test/TagPicker-mount.html">2663</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>TeachingBubble</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25499/perf-test/TeachingBubble-mount_base.html">88881</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25499/perf-test/TeachingBubble-mount.html">93579</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Text</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25499/perf-test/Text-mount_base.html">761</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25499/perf-test/Text-mount.html">810</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>TextField</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25499/perf-test/TextField-mount_base.html">1573</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25499/perf-test/TextField-mount.html">1572</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>ThemeProvider</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25499/perf-test/ThemeProvider-mount_base.html">1485</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25499/perf-test/ThemeProvider-mount.html">1437</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>ThemeProvider</td>
            <td>virtual-rerender</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25499/perf-test/ThemeProvider-virtual-rerender_base.html">1053</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25499/perf-test/ThemeProvider-virtual-rerender.html">1074</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>ThemeProvider</td>
            <td>virtual-rerender-with-unmount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25499/perf-test/ThemeProvider-virtual-rerender-with-unmount_base.html">2138</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25499/perf-test/ThemeProvider-virtual-rerender-with-unmount.html">2129</a></td>
            <td>5000</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25499/perf-test/ThemeProvider-virtual-rerender-with-unmount.regression.txt">Possible regression</a></td>
           </tr>
<tr>
            <td>Toggle</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25499/perf-test/Toggle-mount_base.html">1132</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25499/perf-test/Toggle-mount.html">1099</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>buttonNative</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25499/perf-test/buttonNative-mount_base.html">510</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25499/perf-test/buttonNative-mount.html">530</a></td>
            <td>5000</td>
            <td></td>
           </tr></table></p></details>

<div id="perfComment9423"/>

File: packages/react/src/components/ContextualMenu/ContextualMenu.base.tsx (hunk 1 above)
Comment (micahgodbolt) [reply packages/react/src/components/ContextualMenu/ContextualMenu.base.tsx:713]: oooooh, good catch sarah. i'll double check where that ref is getting placed.

File: packages/react/src/components/ContextualMenu/ContextualMenu.types.ts [hunk 2]
```diff
@@ -47,6 +47,7 @@ export interface IContextualMenuProps
   /**
    * Optional callback to access the IContextualMenu interface. Use this instead of ref for accessing
    * the public methods and properties of the component.
+   * @deprecated ContextualMenu has no public methods, so componentRef no longer returns a ref
```
Comment (khmakoto) [inline packages/react/src/components/ContextualMenu/ContextualMenu.types.ts:50]: This is, technically, not true, it just returns a `ref` to an empty object, doesn't it?

```suggestion
   * @deprecated ContextualMenu has no imperative methods in its public interface, so componentRef returns a useless ref to an empty object
```

Comment (micahgodbolt) [reply packages/react/src/components/ContextualMenu/ContextualMenu.types.ts:50]: pretty sure it returns undefined right now

Comment (micahgodbolt) [reply packages/react/src/components/ContextualMenu/ContextualMenu.types.ts:50]:
```suggestion
   * @deprecated ContextualMenu has no imperative methods in its public interface, so componentRef no longer returns a ref
```

**Review by smhigley — APPROVED** [review]

**Comment by fabricteam** [comment]
## 🕵 fluentuiv8 No visual regressions between this PR and main
<div id="vrtCommentfluentuiv8"/>

**Pull request merged**